upath/core.py

```diff
@@ -38,6 +38,16 @@
 from upath.types import UPathParser
 from upath.types import WritablePathLike
 
+try:
+    import pydantic
+    from pydantic_core import core_schema
+
+    _PYDANTIC_2_AVAILABLE = (
+        (2,) <= tuple(map(int, pydantic.__version__.split(".")[:3])) < (3,)
+    )
+except (ImportError, AttributeError):
+    _PYDANTIC_2_AVAILABLE = False
```

Comment:
Please move the imports into the classmethod, and guard the CoreSchema and GetCoreSchemaHandler types in the TYPE_CHECKING block.